cookbook/integrations/kubernetes/kftensorflow/tf_mnist.py

```diff
@@ -0,0 +1,255 @@
+# * :py:class:`~tensorflow:tf.keras.callbacks.LearningRateScheduler` to decay the learning rate
+def train_model(
+    model: tf.keras.Model,
+    train_dataset: tf.data.Dataset,
+    hyperparameters: Hyperparameters,
+) -> Tuple[tf.keras.Model, str]:
+    # define the checkpoint directory to store the checkpoints
+    checkpoint_dir = "./training_checkpoints"
+
+    # define the name of the checkpoint files
+    checkpoint_prefix = os.path.join(checkpoint_dir, "ckpt_{epoch}")
+
+    # define a callback for printing the learning rate at the end of each epoch
+    class PrintLR(tf.keras.callbacks.Callback):
+        def on_epoch_end(self, epoch, logs=None):
+            print(
+                "\nLearning rate for epoch {} is {}".format(
+                    epoch + 1, model.optimizer.lr.numpy()
+                )
+            )
+
+    # put all the callbacks together
+    callbacks = [
+        tf.keras.callbacks.TensorBoard(log_dir="./logs"),
+        tf.keras.callbacks.ModelCheckpoint(
+            filepath=checkpoint_prefix, save_weights_only=True
+        ),
+        tf.keras.callbacks.LearningRateScheduler(decay),
+        PrintLR(),
+    ]
+
+    # train the model
+    model.fit(train_dataset, epochs=hyperparameters.epochs, callbacks=callbacks)
+
+    # save the model
+    model.save(MODEL_FILE_PATH, save_format="tf")
+
+    return model, checkpoint_dir
+
+
+# %%
+# Evaluation
+# ==========
+#
+# We define ``test_model`` to evaluate loss and accuracy on the test dataset.
+def test_model(
+    model: tf.keras.Model, checkpoint_dir: str, eval_dataset: tf.data.Dataset
+) -> Tuple[float, float]:
+    model.load_weights(tf.train.latest_checkpoint(checkpoint_dir))
+
+    eval_loss, eval_acc = model.evaluate(eval_dataset)
+
+    return eval_loss, eval_acc
+
+
+# %%
+# Defining an MNIST TensorFlow Task
+# ==================================
+#
+# We initialize compute requirements and task output signature. Next, we define a ``mnist_tensorflow_job`` to kick off the training and evaluation process.
+# The task is initialized with ``TFJob`` with certain values set:
+#
+# * ``num_workers``: integer determining the number of worker replicas to be spawned in the cluster for this job
+# * ``num_ps_replicas``: number of parameter server replicas to use
+# * ``num_chief_replicas``: number of chief replicas to use
+#
+# MirroredStrategy uses an all-reduce algorithm to communicate the variable updates across the devices.
+# Hence, ``num_ps_replicas`` is not useful in our example.
+#
+# .. note::
+#   If you'd like to understand the various Tensorflow strategies in distributed training, refer to the `Types of strategies <https://www.tensorflow.org/guide/distributed_training#types_of_strategies>`__ section in the TensorFlow documentation.
+training_outputs = NamedTuple(
+    "TrainingOutputs", accuracy=float, loss=float, model_state=FlyteDirectory
+)
+
+if os.getenv("SANDBOX") != "":
```

Comment:
do we really expose this?

Reply:
I think so because if we are using a sandbox, we'd need to initialize a different set of compute resources. Non-sandbox compute wouldn't work.

Reply:
@samhita-alla who will set the env variable in pod ?

Reply:
it's a good idea to define env specific resource limits/request. We can use `yaml` config for both env(sandbox/cloud), by default we will release the flytesnacks with sandbox config but user can use fast register for overriding the config.  @samhita-alla @kumare3 what do you think ?

Reply:
> who will set the env variable in pod ?

Does it not get automatically set on creating a sandbox?

> it's a good idea to define env specific resource limits/request. We can use yaml/json config for both env(sandbox/cloud), by default we will release the flytesnacks with sandbox config but user can use fast register for overriding the config. @samhita-alla @kumare3 what do you think ?

Do we need to initialize YAML for every code snippet where the resources need to be modified?

Reply:
yes, we currently do this in the pytorch example as well:
https://docs.flyte.org/projects/cookbook/en/latest/auto/case_studies/ml_training/mnist_classifier/pytorch_single_node_multi_gpu.html#defining-the-task

Reply:
> yes, we currently do this in the pytorch example as well: https://docs.flyte.org/projects/cookbook/en/latest/auto/case_studies/ml_training/mnist_classifier/pytorch_single_node_multi_gpu.html#defining-the-task

@cosmicBboy who set that env variable in execution pod

Reply:
> Do we need to initialize YAML for every code snippet where the resources need to be modified?

@samhita-alla  It's just a suggestion if we don't have that env variable

Reply:
@evalsocket, we can propose a standard way of writing sandbox and non-sandbox config, for which we can have a separate PR. What do you think?